be/src/olap/row.h

```diff
@@ -59,6 +59,10 @@ bool equal_row(const std::vector<uint32_t>& ids,
 template<typename LhsRowType, typename RhsRowType>
 int compare_row(const LhsRowType& lhs, const RhsRowType& rhs) {
     for (uint32_t cid = 0; cid < lhs.schema()->num_key_columns(); ++cid) {
+        //because the num_column_ids include the column of double/float type
```

Comment:
This function is used in many senarios. And changing here to fix the problem is not a good idea.
Why not to make column in the schema is not NULL.

Reply:
Ignore the column of double/float type in engine_checksum_task.cpp,  but num_key_columns  include the double/float column index. 

In other ways, I can only add related functions such as reader.init, compare function for sort and aggregation function for checksum. The code volume is a little large.

```    
// ignore float and double type considering to precision lose
    for (size_t i = 0; i < tablet->tablet_schema().num_columns(); ++i) {
        FieldType type = tablet->tablet_schema().column(i).type();
        if (type == OLAP_FIELD_TYPE_FLOAT || type == OLAP_FIELD_TYPE_DOUBLE) {
            continue;
        }

        reader_params.return_columns.push_back(i);
    }
```

Reply:
1. remove the code of ignore the column of float/double type's in engine_checksum.cpp
2. ignore the cell of float/double type when compute row's hash code
3. the value of checksum is independent of the sorting of data rows